### PR TITLE
A0-2935: Investigate flaky e2e tests

### PR DESCRIPTION
**Pull request by ggawryal**
# Description

Changes in e2e-tests and aleph-client only.
* Improved finality version change test [fixes A0-2972] - now it schedules change in (current_session+1) for (current_session+3) instead of in the 1st session for 4th session - it requires one session less and should pass even when ran after 2nd session.
* More checks and better error for era payout test [partially fixes A0-2973] - previously a problem could occur, where one validator was not producing blocks, and because of that session durations were longer than expected. Added block timestamp check to detect such a situation, and report better error message int that case.
* Add logging for finalization test [to make further investigation on A0-2974 easier]

The actual problem in failing e2e-tests is that the chain might sometimes launch incorrectly, with some validators not being able to connect to other peers, causing a finalization stall. One possible solution to this I could imagine is to run the finalization test just after launching the chain and respawn the chain if the test timed out after 60s. Let me know what do you think about it.

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- I have added tests
- I have bumped aleph-client version if relevant

### CODE REVIEW

File: e2e-tests/src/test/era_payout.rs [hunk 1]
```diff
@@ -19,32 +23,45 @@ pub async fn era_payouts_calculated_correctly() -> anyhow::Result<()> {
...
-    let end = expected_payout + one_block;
-    let within_delta = start <= payout && payout <= end;
-    assert!(
-        within_delta,
-        "payout should fall within range: [{start}, {end}] but was {payout}"
-    );
-}
-
-async fn wait_to_second_era<C: StakingApi + WaitingExt>(connection: &C) -> EraIndex {
+async fn wait_to_kth_era<C: StakingApi + WaitingExt>(
```
[the remark this answers is not in the document]
Comment (ggawryal) [reply e2e-tests/src/test/era_payout.rs]: Indeed, replaced

File: aleph-client/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "aleph_client"
-version = "3.5.0"
+version = "3.5.1"
```
Comment (timorl) [inline aleph-client/Cargo.toml:3]: You are adding at least one new functionality (timestamps), so bump the minor version.

**Review by timorl — APPROVED** [review]
Cool, nice to have less flaky tests even if not yet ideal.

**Review by maciejzelaszczyk — APPROVED** [review]
Nice effort!

File: aleph-client/Cargo.toml (hunk 2 above)
Comment (ggawryal) [reply aleph-client/Cargo.toml]: Done

**Pull request merged**